spec/ics-003-connection-semantics/README.md

```diff
@@ -331,20 +328,19 @@ function connOpenTry(
   proofConsensus: CommitmentProof,
   proofHeight: Height,
   consensusHeight: Height) {
-    abortTransactionUnless(validateConnectionIdentifier(desiredIdentifier))
+    // generate a new identifier if the passed identifier was the sentinel empty-string
+    if (desiredIdentifier === "") {
```

Comment:
we cannot do this since the code cannot distinguish between a relayer supplied identifier and a generated identifier without extra logic. I'd suggest changing the if to be more so:
```
if desiredIdentifier != "" {
    // check that previous connection exists
    // check that previous connection fields match
} else {
    identifier = generateIdentifier()
}
```

Reply:
I'd also recommend changing `desiredIdentifier` to `previousIdentifier`

Reply:
see [implementation changes](https://github.com/cosmos/cosmos-sdk/pull/7993/commits/e671648a529439eed3298b5f76e09128a0c6a865#diff-e07c7f16269114e9703a596567c509dfa867968baf0226e46e2b78bf672f8088R74)

Reply:
Yes, I think this `if (desiredIdentifier === "") {` block should move further down, after all validations are done and were a `previous` has been determined. Also, this will prevent an identifier to be generated if checks below fail.
(see https://github.com/cosmos/ics/pull/493#discussion_r527056023)

Reply:
> we cannot do this since the code cannot distinguish between a relayer supplied identifier and a generated identifier without extra logic

Hmm, why would we need to do this? Empty-string is not a valid identifier, it's a sentinel value.

> see implementation changes

I think this structure is fine too, we can update the spec to reflect it if you think it's cleaner, it looks cleaner to me too.

> I'd also recommend changing desiredIdentifier to previousIdentifier

Sure.

> Yes, I think this if (desiredIdentifier === "") { block should move further down, after all validations are done and were a previous has been determined. Also, this will prevent an identifier to be generated if checks below fail.
(see #493 (comment))

It doesn't matter if an identifier is generated and checks later fail, because the transaction is atomic.

Reply:
> Hmm, why would we need to do this? Empty-string is not a valid identifier, it's a sentinel value.

The case I'm thinking of is if the relayer passed in `myownconnectionid` as the connection identifier (a connection that doesn't exist), then the previous would equal null and pass all the checks and continue with the handshake.

Reply:
I've made changes to reflect your suggestion in https://github.com/cosmos/ics/pull/493/commits/43bbbb32fe7b7c2ae0fd641e647a16e4c954f8ed @colin-axner

Reply:
> The case I'm thinking of is if the relayer passed in myownconnectionid as the connection identifier (a connection that doesn't exist), then the previous would equal null and pass all the checks and continue with the handshake.

Ah, I see - yes, that's right.

Reply:
(we could also change the validation functions but your proposal is cleaner)